cfg/builder/builder_finalize.cc

```diff
@@ -239,8 +239,25 @@ void CFGBuilder::removeDeadAssigns(core::Context ctx, const CFG::ReadsAndWrites
                 continue;
             }
 
-            auto fnd = RnW.reads.find(bind.bind.variable);
```

Comment:
this is the actual change in behavior.
It used to see if the variable is "ever" read.
Now instead, we see if the variable is read in _this_ block or is consumed by the outgoing blocks.